### PR TITLE
Depend on information_schema for MyXQL driver

### PR DESCRIPTION
**Pull request by HammamSamara**
This pull request remove the dependency on the tenants table in favor of the existing `information_schema.schemata` table, which brings MySQL tenancy creation flow closer to the postgres flow.

### CODE REVIEW

File: lib/triplex/config.ex [hunk 1]
```diff
@@ -5,9 +5,9 @@ defmodule Triplex.Config do
   - `repo`: the ecto repo that will be used to execute the schema operations.
   - `tenant_prefix`: a prefix for all tenants.
   - `reserved_tenants`: a list of reserved tenants, which cannot be created
-  thourhg triplex APIs. The items here can be strings or regexes.
-  - `tenant_field`: an atom with the name of the field to get the tenant name
-  if the given tenant is a struct. By default it's `:id`.
+  through triplex APIs. The items here can be strings or regexes.
+  - `opts`: extra options to supply for the create database query for MySQL driver
+  supported options are `charset` and `collate`.
```
Comment (kelvinst) [inline lib/triplex/config.ex:10]: I would rather call this `mysql`, as `opts` gives the impression that it is a generic config.

**Review by kelvinst — COMMENTED** [review]
Sorry it took me so long to review, I left you a comment, and I still need to test it manually on my side, to be sure backwards compatibility is maintained.

File: lib/triplex/config.ex [hunk 2]
```diff
@@ -6,7 +6,8 @@ defmodule Triplex.Config do
   - `tenant_prefix`: a prefix for all tenants.
   - `reserved_tenants`: a list of reserved tenants, which cannot be created
   through triplex APIs. The items here can be strings or regexes.
-  - `opts`: extra options to supply for the create database query for MySQL driver
+  - `mysql`: extra options to supply for the create database query for MySQL driver.
+  The SQL standard allows a DEFAULT CHARACTER SET clause in CREATE SCHEMA than are presently accepted by PostgreSQL.
```
Comment (kelvinst) [inline lib/triplex/config.ex:10]: Sorry, so it's possible to configure that for postgres too? 🤔 Not sure if I understand this sentence.

**Comment by HammamSamara** [comment]
> Sorry it took me so long to review, I left you a comment, and I still need to test it manually on my side, to be sure backwards compatibility is maintained.

Thank you for taking the time reviewing the change set. As you may anticipate it, getting rid of the tenants table might be a breaking change for some that needs to be reflected on the repo mix versioning.

File: lib/triplex/config.ex (hunk 2 above)
Comment (HammamSamara) [reply lib/triplex/config.ex:10]: It's actually quoted from compatability section here from Postgres docs https://www.postgresql.org/docs/9.3/sql-createschema.html

> The SQL standard allows a DEFAULT CHARACTER SET clause in CREATE SCHEMA, as well as more subcommand types than are presently accepted by PostgreSQL.

Per my understanding, it means that those options are not supported within the create schema statement for postgres despite being considered as SQL standard.

**Comment by kelvinst** [comment]
Yeah, but I think we could make it backward compatible by rollbacking the `tenants_table` config and using the old code for when the config exists. So we should just change the default to `information_schema` and notify users that when upgrading, they should either run a migration to send all the tenants from the tenants table to the `information_schema` or set the `tenants_table` config and keep using the same as before.

**Comment by HammamSamara** [comment]
Rolling back most of the changes to keep alignment with the tenants table, while using `information_schema` as the default config.
Users need to set `tenants_table` as `:tenants` like I do for tests to keep it backward compatible:
 > config :triplex, tenant_table: :tenants

**Comment by HammamSamara** [comment]
After a quick thought about compatibility and for safer merge, I am setting the default behavior to work with the `:tenants` table so all users segments don't need to change anything now (100% backward compatible).

If anyone (like me) wishes to not work with `:tenant` table and generates the associate migrations, then they can just configure Triplex to work with the "information_schema.schemata" as mentioned in the docs.
> config :triplex, tenant_table: :"information_schema.schemata"

File: lib/triplex/config.ex (hunk 2 above)
Comment (kelvinst) [reply lib/triplex/config.ex:12]: Well, if I understand correctly, it is saying that it does accept, so the `charset` option could be a generic one. But hey, why introducing both these options in this PR? Why not moving this change to another one?

Comment (HammamSamara) [reply lib/triplex/config.ex]: Agreed. Just dropped those options to be introduced in different pr.

**Comment by kelvinst** [comment]
Pretty cool! Awesome work, thanks! And sorry for all the back and forth. Merging it to master, will release a new version of triplex with your changes soon.

**Review by kelvinst — APPROVED** [review]

**Comment by kelvinst** [comment]
@HammamSamara one last request: could you open a new PR changing the default behavior to use the information schema? I  think that's a sane default as it's simpler than having to add a new table. We will then release that other one on 2.0

**Comment by HammamSamara** [comment]
PR opened here https://github.com/ateliware/triplex/pull/81

**Pull request merged**